app/main/data.py

```diff
@@ -0,0 +1,54 @@
+import json
+import os
+
+import requests
+from flask import current_app, send_file
+
+
+def get_remote_data(endpoint):
+    """
+        Queries the api endpoint provided and returns a
+        data response in json format.
+
+    Args:
+        endpoint (str): an API get data address
+
+    Returns:
+        data (json): data response in json format
+    """
+
+    response = requests.get(endpoint)
+    if response.status_code == 200:
+        data = response.json()
+        return data, 200
+    else:
+        current_app.logger.warn(
+            "GET remote data call was unsuccessful with status code:"
+            f" {response.status_code}."
+        )
+        return None, response.status_code
+
+
+def dwnload_json_file(resp, download_path):
+    """
+    Retrieves json object from response.
+
+    Args:
+        resp (json): data response in json format
+        download_path (str):
+
+    Returns:
+        send_file() (function): a function that downloads the json data as a json file
+    """
+
+    filename = "downloaded_data.json"
+    folder = os.path.join(current_app.root_path, download_path)
+    filepath = os.path.join(folder, filename)
+
+    if not os.path.exists(folder):
+        os.mkdir(folder)
+
+    with open(filepath, "w") as file:
+        json.dump(resp, file)
```

Comment:
Can we not write this to disk, we should read it to memory in a file like object instead.